blogs/vonage-solaria-python-fastapi/src/server.py

```
@@ -0,0 +1,449 @@
+
+def create_session(force=False):
+    """Initialize a Gladia real-time transcription session with exponential backoff retry."""
+    # If we already have a session and not forcing a new one, return it
+    if gladia_session["url"] and not force:
+        return gladia_session["url"]
+    
+    # Rate limiting - wait at least 2 seconds between attempts
+    current_time = time.time()
+    time_since_last_attempt = current_time - gladia_session["last_init_attempt"]
+    if time_since_last_attempt < 2 and gladia_session["last_init_attempt"] > 0:
+        logger.debug(f"Rate limiting: waiting before retrying Gladia session creation")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)